task/control-plane-agent/src/mgs_common.rs

```diff
@@ -143,6 +147,108 @@ impl MgsCommon {
+        if self.reset_component_requested != component {
+            return Err(SpError::ResetComponentTriggerWithoutPrepare);
+        }
+        // If we are not resetting the SP_ITSELF, then we may come back here.
+        self.reset_component_requested.id.fill(0);
+
+        // For now, resetting the SP through reset_component() is
+        // the same as through reset()
+        if component == SpComponent::SP_ITSELF {
+            task_jefe_api::Jefe::from(crate::JEFE.get_task_id())
+                .request_reset();
+            // If `request_reset()` returns,
+            // something has gone very wrong.
+            panic!();
+        }
+
+        // mgs_{gimlet,psc,sidecar}.rs deal with any board specific
+        // reset strategy. Here we take care of common SP and RoT cases.
+        let target = if matches!(
+            intent,
+            ResetIntent::Persistent | ResetIntent::Transient
+        ) {
+            match component {
+                SpComponent::ROT => match slot {
+                    Some(0) => UpdateTarget::ImageA,
+                    Some(1) => UpdateTarget::ImageB,
+                    _ => return Err(SpError::RequestUnsupportedForComponent),
+                },
+                _ => return Err(SpError::RequestUnsupportedForComponent),
+            }
+        } else {
+            // TODO: This value is ignored when intent is Normal or Expensive*
+            UpdateTarget::ImageA
+        };
+        match update.sprot_task().reset_component(
+            intent.into(),
+            target,
+            auth_data.len() as u16,
+            auth_data,
+        ) {
+            Err(SprotError::RspTimeout) => {
+                // This is the expected error if the reset was successful.
+                // It could be that the RoT is out-to-lunch for some other
+                // reason though.
+                // Things for upper layers to do:
+                // TODO: Check boot nonce to see if we are in a new session.
+                // TODO: Check that the expected image is now running.
+                // (Management plane should do that.)
+                // TODO: Enable staged updates where we don't automatically
+                // reset after writing an image.
+                ringbuf_entry!(Log::RotReset {
+                    err: SprotError::RspTimeout
+                });
+            }
+            Err(err) => {
+                // Some other error occurred.
+                // TODO: Update is all-or-nothing at the moment.
+                // The control plane can try to reset the RoT again or it
+                // can start the update process all over again.  We should
+                // be able to make incremental progress if there is some
+                // bug/condition that is degrading SpRot communications.
+                ringbuf_entry!(Log::RotReset { err });
+                // TODO: send an error back. However, changes need to be
```

Comment:
This TODO seems pretty darn significant.... With the current code, how can we tell if an error occurred here?